fuse-android/src/main/java/com/github/kittinunf/fuse/android/MemLruCache.kt

```diff
@@ -1,43 +1,35 @@
 package com.github.kittinunf.fuse.android
 
 import android.util.LruCache
-import com.github.kittinunf.fuse.core.cache.KeyType
+import com.github.kittinunf.fuse.core.cache.Entry
 import com.github.kittinunf.fuse.core.cache.Persistence
 
 class MemLruCache(private val cache: LruCache<String, Any>) : Persistence<Any> {
 
-    override fun put(safeKey: String, key: String, value: Any, timeToPersist: Long) {
-        cache.apply {
-            put(safeKey, value)
-            put(convertKey(safeKey, KeyType.Key.ordinal.toString()), key)
-            put(convertKey(safeKey, KeyType.Time.ordinal.toString()), timeToPersist)
-        }
+    override fun put(safeKey: String, entry: Entry<Any>) {
+        cache.put(safeKey, entry)
     }
 
-    override fun remove(safeKey: String): Boolean {
-        val removedValue = cache.remove(safeKey)
-        cache.remove(convertKey(safeKey, KeyType.Key.ordinal.toString()))
-        cache.remove(convertKey(safeKey, KeyType.Time.ordinal.toString()))
-        return removedValue != null
-    }
+    override fun remove(safeKey: String): Boolean = cache.remove(safeKey) != null
 
     override fun removeAll() {
         cache.evictAll()
     }
 
     override fun allKeys(): Set<String> {
-        return cache.snapshot().keys.filter { !it.contains(".") }
-            .map { get(convertKey(it, KeyType.Key.ordinal.toString())) as String }
-            .toSet()
+        return synchronized(this) {
+            val snapshot = cache.snapshot()
+            snapshot.keys.map { getEntry(it)!!.key }.toSet()
+        }
     }
 
     override fun size(): Long {
         return cache.size().toLong()
     }
 
```

Comment:
[nit] unrelated, but maybe `size` and `allKeys` could be nicer as a property instead?

Reply:
You commented a bit too late lol! This could break the client that consumes the library from the previous version though ...

Reply:
That's true, maybe the timing could be during a major update